# HG changeset patch
# User Mike Miller <mtmiller@octave.org>
# Date 1502119168 25200
#      Mon Aug 07 08:19:28 2017 -0700
# Branch stable
# Node ID 6ae0292aaa5c3a43db197076db82541831d5a204
# Parent  5b5b021b1476f75aa1e5cce53a19275eb5f0ce50
configure: ensure empty pkg-config results are actually empty (bug #51680)

* configure.ac (MAGICK_CPPFLAGS, MAGICK_LDFLAGS, MAGICK_LIBS): Trim pkg-config
results to eliminate strings that are entirely spaces.  This is a workaround
for versions of pkg-config older than 0.28 that may emit an answer that looks
empty but actually contains space characters.
* acinclude.m4 (OCTAVE_CHECK_LIB, OCTAVE_CHECK_QT_VERSION): Likewise.

diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -1559,19 +1559,19 @@ if test $check_magick = yes; then
   MAGICK_LDFLAGS=
   MAGICK_LIBS=
 
   PKG_CHECK_EXISTS([$magick++], [
     ## Make sure we only get -I, -L, and -l flags.  Some Graphics/ImageMagick++
     ## packages add extra flags that are useful when building
     ## Graphics/ImageMagick++ extentions.  These extra flags break the
     ## Octave build.
-    MAGICK_CPPFLAGS=`$PKG_CONFIG --cflags-only-I $magick++`
-    MAGICK_LDFLAGS=`$PKG_CONFIG --libs-only-L $magick++`
-    MAGICK_LIBS=`$PKG_CONFIG --libs-only-l $magick++`
+    MAGICK_CPPFLAGS="$($PKG_CONFIG --cflags-only-I $magick++ | $SED -e 's/^ *$//')"
+    MAGICK_LDFLAGS="$($PKG_CONFIG --libs-only-L $magick++ | $SED -e 's/^ *$//')"
+    MAGICK_LIBS="$($PKG_CONFIG --libs-only-l $magick++ | $SED -e 's/^ *$//')"
 
     warn_magick="$magick++ library fails tests.  The imread, imwrite, and imfinfo functions for reading and writing image files will not be fully functional."
 
     save_CPPFLAGS="$CPPFLAGS"
     save_LIBS="$LIBS"
     CPPFLAGS="$MAGICK_CPPFLAGS $CPPFLAGS"
     LIBS="$MAGICK_LDFLAGS $MAGICK_LIBS $LIBS"
     AC_LANG_PUSH(C++)
diff --git a/m4/acinclude.m4 b/m4/acinclude.m4
--- a/m4/acinclude.m4
+++ b/m4/acinclude.m4
@@ -686,22 +686,22 @@ AC_DEFUN([OCTAVE_CHECK_LIB], [
     *)
       m4_toupper([$1])_LIBS="-l$with_$1"
     ;;
   esac
 
   if test $ac_octave_$1_pkg_check = yes; then
     PKG_CHECK_EXISTS(m4_default([$9], [$1]), [
       if test -z "$m4_toupper([$1])_CPPFLAGS"; then
-        m4_toupper([$1])_CPPFLAGS="$($PKG_CONFIG --cflags-only-I m4_default([$9], [$1]))"
+        m4_toupper([$1])_CPPFLAGS="$($PKG_CONFIG --cflags-only-I m4_default([$9], [$1]) | $SED -e 's/^ *$//')"
       fi
       if test -z "$m4_toupper([$1])_LDFLAGS"; then
-        m4_toupper([$1])_LDFLAGS="$($PKG_CONFIG --libs-only-L m4_default([$9], [$1]))"
+        m4_toupper([$1])_LDFLAGS="$($PKG_CONFIG --libs-only-L m4_default([$9], [$1]) | $SED -e 's/^ *$//')"
       fi
-      m4_toupper([$1])_LIBS="$($PKG_CONFIG --libs-only-l m4_default([$9], [$1]))"
+      m4_toupper([$1])_LIBS="$($PKG_CONFIG --libs-only-l m4_default([$9], [$1]) | $SED -e 's/^ *$//')"
     ])
   fi
 
   if test -n "$m4_toupper([$1])_LIBS"; then
     ac_octave_save_CPPFLAGS="$CPPFLAGS"
     ac_octave_save_LDFLAGS="$LDFLAGS"
     ac_octave_save_LIBS="$LIBS"
     CPPFLAGS="$m4_toupper([$1])_CPPFLAGS $CPPFLAGS"
@@ -1355,19 +1355,19 @@ AC_DEFUN([OCTAVE_CHECK_QT_VERSION], [AC_
 
   PKG_CHECK_MODULES(QT, [$QT_MODULES],
     [],
     [build_qt_gui=no
      warn_qt_libraries="Qt libraries not found; disabling Qt GUI"])
 
   if test $build_qt_gui = yes; then
     ## Retrieve Qt compilation and linker flags
-    QT_CPPFLAGS="`$PKG_CONFIG --cflags-only-I $QT_MODULES`"
-    QT_LDFLAGS="`$PKG_CONFIG --libs-only-L $QT_MODULES`"
-    QT_LIBS="`$PKG_CONFIG --libs-only-l $QT_MODULES`"
+    QT_CPPFLAGS="$($PKG_CONFIG --cflags-only-I $QT_MODULES | $SED -e 's/^ *$//')"
+    QT_LDFLAGS="$($PKG_CONFIG --libs-only-L $QT_MODULES | $SED -e 's/^ *$//')"
+    QT_LIBS="$($PKG_CONFIG --libs-only-l $QT_MODULES | $SED -e 's/^ *$//')"
 
     case $host_os in
       *darwin*)
         ## Qt might be installed in framework
         if test -z "$QT_LIBS"; then
           QT_LDFLAGS="`$PKG_CONFIG --libs-only-other $QT_MODULES | tr ' ' '\n' | $GREP -e '-F' | uniq | tr '\n' ' '`"
           QT_LIBS="`$PKG_CONFIG --libs-only-other $QT_MODULES | tr ' ' '\n' | $GREP -v -e '-F' | uniq | tr '\n' ' '`"
           ## Enabling link_all_deps works around libtool's imperfect handling
